client/src/components/ReviewForm.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom"
+
+import translateServerErrors from "../services/translateServerErrors.js"
+
+const ReviewForm = (props) => {
+
+    const [newReview, setNewReview] = useState({
+        textBody: "",
+        rating: ""
+    })
+
+    const [errors, setErrors] = useState({})
+    const site = props.site
+
+    const addNewReview = async (formData) => {
+        try {
+            const response = await fetch(`/api/v1/sites/${site.id}`, {
+                method: "POST",
+                headers: new Headers({
+                    "Content-Type": "application/json",
+                }),
+                body: JSON.stringify(formData)
+            })
+
+            if (!response.ok) {
+                if (response.status === 422) {
+                    const body = await response.json()
+                    const newErrors = translateServerErrors(body.errors)
+                    return setErrors(newErrors)
+                } else {
+                    throw (new Error(`${response.status} (${response.statusText})`))
+                }
+            } else {
+                const responseBody = await response.json()
+                const reviewData = site.reviews.concat(responseBody.newReview)
+                setErrors({})
+                props.setSite({ ...site, reviews: reviewData })
+            }
+        } catch (error) {
+            console.error(`Error in fetch: ${error.message}`)
+        }
+    }
+
+    const handleInputChange = (event) => {
+        setNewReview({
+            ...newReview,
+            [event.currentTarget.name]: event.currentTarget.value
+        })
+    }
+    
+    const clearForm = () => {
+        setNewReview({
+            textBody: "",
+            rating: ""
+        })
+    }
+
+    const currentUser = props.currentUser
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        if (currentUser) {
+            addNewReview(newReview)
+        }
+    }
```

Comment:
while this is okay, it would be much better to not render a review form in the first place! 
consider: a non-signed-user is like "dope, I want to review" and they submit the form...and nothing happens. Frustrating!

Reply:
very good point!